students/JeroenSweerts/Lesson3/mailroom.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+donors = ['papa', 'mama', 'bompa', 'bobonne', 'onbekende']
+donations = [[100, 5, 15], [12, 200, 2, 66], [1000], [500, 500], [1000000]]
+
+def thankyou():
+    '''If the user types ‘list’, show them a list of the donor names and re-prompt
+    If the user types a name not in the list, add that name to the data structure
+    and use it.
+    If the user types a name in the list, use it.
+    Once a name has been selected, prompt for a donation amount.
+    Turn the amount into a number – it is OK at this point for the program to
+    crash if someone types a bogus amount.
+    Once an amount has been given, add that amount to the donation history of
+    the selected user.
+    Finally, use string formatting to compose an email thanking the donor for their
+    generous donation. Print the email to the terminal and return to the original
+    prompt.'''
+    name = input('Please enter a name or type "list" to see a list of names > ')
+    while name == 'list':
```

Comment:
In Python, you can initialize an infinite loop by using the syntax `while True:` for these kinds of situations and then use `break` to exit the loop whenever your exit condition is met. 

Or you can set up a Boolean variable such as `asking_for_name = True` for a more readable version of the same construct in `while asking_for_name`. 

One of the advantages of using an approach like this here is that you wouldn't need to repeat the same call to `input` at various points in your code. So the whole block would look something like this:

```
    while True:
        name = input('Please enter a name or type "list" to see a list of names > ')
        if name == 'list':
            print(donors)
        else:
            break
```